src/ui/public/courier/fetch/call_response_handlers.js

```diff
@@ -35,11 +34,17 @@ export function CallResponseHandlersProvider(Private, Promise) {
       const response = responses[index];
 
       if (response.timed_out) {
-        courierNotifier.warning(new SearchTimeout());
+        toastNotifications.addWarning({
+          title: 'The data shown may be incomplete',
+          text: 'All or part of your request has timed out',
+        });
       }
 
       if (response._shards && response._shards.failed) {
-        courierNotifier.warning(new ShardFailure(response));
+        toastNotifications.addWarning({
+          title: 'Shard failure',
+          text: `${response._shards.failed} of ${response._shards.total} shards failed`,
```

Comment:
One of the most annoying things with shard failures currently is, that you don't get any insights into the actual failures, and that's very often a painpoint in discuss questions, since users need to check the actual browser request, and dig through that JSON. In the long run we definitely should build this notification in a way, that you can get a list of actual errors from the individual charts. Since you were anyway touching this place, I thought I mention that right now, in case you feel like we should do it within that PR. But I am also totally fine having that in a different PR (but I think we should take care of that rather sooner than later).

Reply:
Great point Tim. I think this is outside of the scope for this PR, but I've updated https://github.com/elastic/kibana/issues/5957 with more information. Hopefully we can move on this sometime soon.

Reply:
Yes, it would be good to add more info about the shard failure.  When you do, it might be good to put this information in something other than a toast because a toast times out after a few seconds.

In the meantime, the toast can simply be one line:

3 of  6 shards failed